### ISSUE TITLE
Support for sidebar control via single root _meta file

### ISSUE THREAD

**Issue by KRRISH96** [shuding/nextra#1239]
I have been trying nextra recently and loving it. Thank you for building this :)

I feel a bit limited in terms of controlling the sidebar,
Currently, I can only structure the items in the sidebar for a particular directory and have to create `_meta` in the nested sub-directory to control respective items. Can easily get hectic as the nested level goes deep and there are many directories and sub-directories as well.

I do understand why managing _meta at a respective directory will make things easier when moving around. But I think a single `_meta` control should also be allowed. Similar to menus where we can add as children.

Similar to [menu with items](https://nextra.site/docs/docs-theme/page-configuration#menus)

```js
{
  "company": {
    "title": "Company",
    "type": "category" // <== maybe a new type as "category"?,
    "items": { // <== Like this
      "about": "About",
      "contact": "Contact Us",
      "policies": {
        "title": "Policies",
          "items": { // <== With Nested items too
            "terms": "Terms of Service",
            "legal": "Legal Policies"
          }
      }
    }
  }
}
```

Any suggestions/workarounds in achieving this would be really helpful. Thank you 🙌

**Comment by KRRISH96** [comment]
Similar behavior. I'm wondering if there is a way to just have one `_meta.json` file at the root to customize (titles, order, etc.,)

Something like this from docusaurus (https://docusaurus.io/docs/sidebar/items#sidebar-item-category).

the use case is I want to have a single source of truth for my sidebar (`docs/_meta.json`) that has options to customize everything from root level to nested doc files and render the sidebar accordingly.

**Comment by shuding** [comment]
In our initial design, _meta.json at each folder level is great when you move docs around. For example when moving an entire folder to another place, everything just works.

Another reason is we want some kind of "dynamic meta" in specific levels. You will see that in our upcoming updates.

**Comment by shuding** [comment]
> But I think a single _meta control should also be allowed. Similar to menus where we can add as children.

Agreed! It's just that we need to be careful about conflicts etc.

**Comment by KRRISH96** [comment]
> In our initial design, _meta.json at each folder level is great when you move docs around. For example when moving an entire folder to another place, everything just works.
> 
> Another reason is we want some kind of "dynamic meta" in specific levels. You will see that in our upcoming updates.


+1 on this approach as it's quite common that docs are moved around, especially with major version changes. However, root-level control should still be allowed IMHO.
Also, I did try the dynamic meta out and liked it 💯 .

**Comment by KRRISH96** [comment]
> > But I think a single _meta control should also be allowed. Similar to menus where we can add as children.
> 
> Agreed! It's just that we need to be careful about conflicts etc.

Yes, conflicts would definitely arise. I think maybe the nested _meta (the one in the child directory) should take precedence over the root_meta. Similar behavior to `!important` in CSS.
So, even if root_meta defines the sidebar structure, a child directory can be overridden with _meta at that particular directory. I will leave the call on this to you experts :)
 
P.S. I would love to contribute to this if it helps.

**Comment by Prottoy2938** [comment]
Any update on this? Is it possible now to use single `_meta.[langCode].json`?

**Comment by Prottoy2938** [comment]
> 

how'd you try out the `dynamic meta`? Where can I find the doc or code for it?

**Comment by Prottoy2938** [comment]
> > > But I think a single _meta control should also be allowed. Similar to menus where we can add as children.
> > 
> > 
> > Agreed! It's just that we need to be careful about conflicts etc.
> 
> Yes, conflicts would definitely arise. I think maybe the nested _meta (the one in the child directory) should take precedence over the root_meta. Similar behavior to `!important` in CSS.
> 
> So, even if root_meta defines the sidebar structure, a child directory can be overridden with _meta at that particular directory. I will leave the call on this to you experts :)
> 
> P.S. I would love to contribute to this if it helps.

> 

how'd you try out the `dynamic meta`? Where can I find the doc or code for it?

**Comment by dimaMachina** [comment]
Available in Nextra 4 https://the-guild.dev/blog/nextra-4#_metaglobal-file
